src/main/java/net/glowstone/net/handler/play/player/UseItemHandler.java

```diff
@@ -90,6 +92,16 @@ static void handleRightClickAir(GlowPlayer player, ItemStack holding, EquipmentS
             if (type != null) {
                 if (type.getContext().isAirApplicable()) {
                     type.rightClickAir(player, holding);
+                } else {
+                    if ((holding.getType() == Material.WATER_BUCKET
```

Comment:
Seems like this was incorrectly reverted from the ender pearl PR.

Reply:
Fixed.